ambry-protocol/src/main/java/com.github.ambry.protocol/CompositeSend.java

```diff
@@ -48,6 +50,13 @@ public long writeTo(WritableByteChannel channel) throws IOException {
     return written;
   }
 
+  @Override
+  public void writeTo(AsyncWritableChannel channel, Callback callback) throws IOException {
+    for (Send send : compositSendList) {
+      send.writeTo(channel, callback);
+    }
+  }
+
```

Comment:
we should only call callback method once for the entire list since the callback passed to writeTo should be invoked only when the write is completed or failed.

Reply:
Yes, perhaps each child send should have its own callback and then when either all of those callbacks succeed or an error has occured, the passed-in callback is notified, like in the default impl of `AsyncWritableChannel.write(ByteBuf src, Callback<Long> callback)`

Reply:
Yes, this should be changed. 
I will make another PR to address this issue. For this PR, I will set callback to null in child send.